src/functional/swing_slide_filter.zig

```diff
@@ -0,0 +1,552 @@
+            try getSwingLine(&current_segment, &current_line, 0.0);
+            try decompressed_values.append(current_segment.start_value);
+            var current_timestamp: usize = current_segment.start_time + 1;
+            while (current_timestamp < current_segment.end_time) : (current_timestamp += 1) {
+                const y: f64 = f80Tof64(evaluate(&current_line, current_timestamp));
+                try decompressed_values.append(y);
+            }
+            try decompressed_values.append(current_segment.end_value);
+            first_timestamp = current_timestamp + 1;
+        } else {
+            try decompressed_values.append(current_segment.start_value);
+            first_timestamp += 1;
+        }
+    }
+}
+
+test "swing-filter single line compress and decompress" {
+    const alloc = testing.allocator;
+    const uncompressed_values = [_]f64{ 1.5, 2.5, 3.5, 4.5, 5.5, 6.5, 7.5 };
+    var compressed_values = ArrayList(u8).init(alloc);
+    defer compressed_values.deinit();
+    var decompressed_values = ArrayList(f64).init(alloc);
+    defer decompressed_values.deinit();
+
+    const error_bound: f32 = 0.0;
+
+    try compress_swing(uncompressed_values[0..], &compressed_values, error_bound);
+
+    try decompress(compressed_values.items, &decompressed_values);
+    try testing.expect(mem.eql(f64, uncompressed_values[0..], decompressed_values.items));
+}
+
+test "swing-filter odd size compress and decompress" {
+    const alloc = testing.allocator;
+    const uncompressed_values = [_]f64{ 1.5, 2.5, 3.5, 4.5, 5.5, 6.5, 7.5, 8.5, 1.5 };
+    var compressed_values = ArrayList(u8).init(alloc);
+    defer compressed_values.deinit();
+    var decompressed_values = ArrayList(f64).init(alloc);
+    defer decompressed_values.deinit();
+
+    const error_bound: f32 = 0.0;
+
+    try compress_swing(uncompressed_values[0..], &compressed_values, error_bound);
+
+    try decompress(compressed_values.items, &decompressed_values);
+
+    try testing.expect(mem.eql(f64, uncompressed_values[0..], decompressed_values.items));
+}
+
+test "swing-filter single point even size compress and decompress" {
+    const alloc = testing.allocator;
+    const uncompressed_values = [_]f64{ 1.5, 2.5, 3.5, 4.5, 5.5, 6.5, 7.5, 8.5, 9.5, 1.5 };
+    var compressed_values = ArrayList(u8).init(alloc);
+    defer compressed_values.deinit();
+    var decompressed_values = ArrayList(f64).init(alloc);
+    defer decompressed_values.deinit();
+
+    const error_bound: f32 = 0.0;
+
+    try compress_swing(uncompressed_values[0..], &compressed_values, error_bound);
+
+    try decompress(compressed_values.items, &decompressed_values);
+    try testing.expect(mem.eql(f64, uncompressed_values[0..], decompressed_values.items));
+}
+
+test "swing-filter two parallel lines compress and decompress" {
+    const alloc = testing.allocator;
+    const uncompressed_values = [_]f64{ 1.5, 2.5, 3.5, 4.5, 5.5, 6.5, 7.5, 8.5, 1.5, 2.5, 3.5, 4.5, 5.5, 6.5, 7.5, 8.5 };
```

Comment:
Maybe place the numbers on separate line(s) as the line seems to [be above 100 characters](https://ziglang.org/documentation/master/#Whitespace).

Reply:
That test was removed completly.